File: DART/DPD/lib/creational/abstract_factory/concrete_log_factory.dart
import 'abstract_log_factory.dart';
import 'abstract_log_product.dart';
import 'concrete_file_log_product.dart';
import 'concrete_terminal_log_product.dart';

final class ConcreteLogFactory implements AbstractLogFactory {
  @override
  AbstractLogProduct makeFile(String path) => ConcreteFileLogProduct(path);

  @override
  AbstractLogProduct makeTerminal() => ConcreteTerminalLogProduct();
}

File: JAVA/DPJ/src/main/java/abstract_factory/ConcreteLogFactory.java
package abstract_factory;

public class ConcreteLogFactory implements AbstractLogFactory {
    @Override
    public AbstractLogProduct makeTerminal() {
        return new ConcreteLogTerminalProduct();
    }

    @Override
    public AbstractLogProduct makeFile() {
        return new ConcreteLogFileProduct();
    }
}
